server/channels/app/slashcommands/command_invite.go

```diff
@@ -59,28 +70,127 @@ func (i *InviteProvider) doCommand(a *app.App, c request.CTX, args *model.Comman
 	// Verify that the inviter has permissions to invite users to the every channel.
 	targetChannels = i.checkPermissions(a, c, args, resps, targetUsers[0], targetChannels)
 
+	// track errors returned for various users.
+	differentChannels := make([]string, 0, 1)
+	nonTeamUsers := make([]string, 0, 1)
+	teamConstrained := make([]*model.User, 0, 1)
+	usersInChannel := make([]string, 0, 1)
+	errorUsers := make([]*model.User, 0, 1)
+
 	for _, targetUser := range targetUsers {
 		for _, targetChannel := range targetChannels {
-			if resp = i.addUserToChannel(a, c, args, targetUser, targetChannel); resp != "" {
-				*resps = append(*resps, resp)
-				continue
+			userError := i.addUserToChannel(a, c, args, targetUser, targetChannel)
+			if userError == NoError {
+				if args.ChannelId != targetChannel.Id {
+					differentChannels = append(differentChannels, targetUser.Username)
+				}
+			} else if userError == UserNotInTeam {
+				nonTeamUsers = append(nonTeamUsers, targetUser.Username)
+			} else if userError == IsConstrained {
+				teamConstrained = append(teamConstrained, targetUser)
```

Comment:
Does this account for both team constrained and channel constrained?

Reply:
Actually only channel constrained...we don't try to add to team, rather show error, cannot be added. I did update the variable to `channelConstrained`, so less confusing.